prov/efa/src/rxr/rxr_read.c

```diff
@@ -543,31 +545,52 @@ int rxr_read_post(struct rxr_ep *ep, struct rxr_read_entry *read_entry)
 		if (OFI_UNLIKELY(!pkt_entry))
 			return -FI_EAGAIN;
 
-		rxr_pkt_init_read_context(ep, read_entry, segsize, pkt_entry);
-
-		ret = fi_read(lower_ep,
-			      iov_ptr, segsize, read_entry->mr_desc[iov_idx],
-			      lower_ep_addr,
-			      (uint64_t)rma_iov_ptr, read_entry->rma_iov[rma_iov_idx].key,
-			      pkt_entry);
+		rxr_pkt_init_read_context(ep, read_entry, iov.iov_len, pkt_entry);
+
+		memset(&msg, 0, sizeof(msg));
+		msg.msg_iov = &iov;
+		msg.desc = &read_entry->mr_desc[iov_idx];
+		msg.iov_count = 1;
+		msg.rma_iov = &rma_iov;
+		msg.rma_iov_count = 1;
+		msg.context = pkt_entry;
+
+		if (read_entry->lower_ep_type == SHM_EP) {
+			msg.addr = shm_fiaddr;
+			ret = fi_readmsg(ep->shm_ep, &msg, 0);
+		} else {
+			efa_ep = container_of(ep->rdm_ep, struct efa_ep, util_ep.ep_fid);
+			msg.addr = read_entry->addr;
+			self_comm = (read_entry->context_type == RXR_READ_CONTEXT_PKT_ENTRY);
```

Comment:
Since `RXR_READ_CONTEXT_PKT_ENTRY`indicates self_comm, I'd like to rename it to indicate self comm better.

Reply:
Not sure how to rename it, because it is defined with ```RXR_READ_CONTEX_TX_ENTRY``` and ```RXR_READ_CONTEXT_RX_ENTRY```.  There is comments about it in rxr_read.h